scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -337,76 +569,28 @@ impl MacroPlugin for ProcMacroHostPlugin {
         item_ast: ast::ModuleItem,
         _metadata: &MacroPluginMetadata<'_>,
     ) -> PluginResult {
-        // Apply expansion to `item_ast` where needed.
-        let expansions = self
-            .handle_attribute(db, item_ast.clone())
-            .into_iter()
-            .chain(self.handle_derive(db, item_ast.clone()));
-        let stable_ptr = item_ast.clone().stable_ptr().untyped();
-        let file_path = stable_ptr.file_id(db).full_path(db.upcast());
-        let file_id = short_hash(file_path.clone());
-
-        let mut token_stream = TokenStream::from_item_ast(db, item_ast)
-            .with_metadata(TokenStreamMetadata::new(file_path, file_id));
-        let mut aux_data = EmittedAuxData::default();
-        let mut modified = false;
-        let mut all_diagnostics: Vec<Diagnostic> = Vec::new();
-        for input in expansions {
-            let result = self
-                .instance(input.package_id)
-                .generate_code(input.expansion.name.clone(), token_stream.clone());
+        let stream_metadata = Self::calculate_metadata(db, item_ast.clone());
+
+        // Expand first attribute.
+        // Note that we only expand the first attribute, as we assume that the rest of the attributes
+        // will be handled by a subsequent call to this function.
+        if let (Some(input), token_stream) = self.parse_attribute(db, item_ast.clone()) {
+            let token_stream = token_stream.with_metadata(stream_metadata.clone());
+            let stable_ptr = item_ast.clone().stable_ptr().untyped();
+            return self.expand_attribute(input, token_stream, stable_ptr);
```

Comment:
Note this takes a bit different approach to what we did before:
We remove the attribute, process it, and return expanded code. 
Following attributes will be processed in a subsequent macro calls.